neat/tools/fittools/compartmentfitter.py

```diff
@@ -65,46 +66,44 @@ def getExpansionPoints(e_hs, channel, only_e_h=False):
     channel: `neat.channels.ionchannels.IonChannel`
         The ion channels
     only_e_h: bool
-        If True, othe entries in ``sv_hs`` are only at ``e_hs``
+        Only applicable for channels with at least two state variables.
+        If True, returned expansion points are always for state variable
+        combination evaluated at the same holding potential. Otherwise,
+        state variable activations are evaluated at different holding potentials.
 
     Returns
     -------
-    sv_hs: list of dict
-        Each entry is a state variable expansion point
-    e_hs: list of float
-        The holding potentials corresponding to each entry in ``sv_hs``
+    sv_hs: dict
+        the expansion points at every holding potential
     """
-    e_hs = list(e_hs)
     if len(channel.statevars) == 1 or only_e_h:
-        sv_hs = [channel.computeVarinf(e_h) for e_h in e_hs]
-    elif len(channel.statevars) == 2:
-        # evaluate at the holding potentials
-        sv_aux = [channel.computeVarinf(e_h) for e_h in e_hs]
-
-        # check which variable is activation
-        sv = channel.computeVarinf(np.array([-43.22, -32.22]))
-        sind_act = None
-        for ii, svar in enumerate(channel.ordered_statevars):
-            if sv[svar][1] > sv[svar][0]:
-                sind_act = 'ii' if ii == 0 else 'jj'
-
-        # evaluate at combinations of holding potentials
-        sv_hs_extra, e_hs_extra = [], []
-        sv_o = channel.ordered_statevars
-        for ii, sv_1 in enumerate(sv_aux):
-            for jj, sv_2 in enumerate(sv_aux):
-                sv_hs_extra.append({str(sv_o[0]): sv_1[sv_o[0]],
-                                    str(sv_o[1]): sv_2[sv_o[1]]})
-                # follow holding potential of activation state variable
-                e_hs_extra.append(eval('e_hs[%s]'%sind_act))
-
-        sv_hs = sv_aux + sv_hs_extra
-        e_hs = e_hs + e_hs_extra
+        sv_hs = channel.computeVarinf(e_hs)
+        sv_hs['v'] = e_hs
     else:
-        raise Exception('Method only implemented for channels with two ' + \
-                        'or less state variables')
-
-    return sv_hs, e_hs
+        # test voltage values to check whether state variable is activating or
+        # inactivating
+        v_test = np.array([-43.22, -32.22])
```

Comment:
maybe move this down just before the loop where it's used? also, could this check be hidden away in a function with an appropriate name?

Reply:
implemented in commit `649895f`